keras2onnx/common/onnx_ops.py

```diff
@@ -346,70 +431,13 @@ def apply_upsample(scope, input_name, output_name, container, operator_name=None
         attrs['width_scale'] = float(scales[3])
         attrs['mode'] = mode.upper()
         op_version = 1
-    else:
+    elif container.target_opset < 9:
         attrs['scales'] = list(map(float, scales))
         attrs['mode'] = mode.lower()
```

Comment:
when target_opset == 9, shall we also  attrs['mode'] = mode.lower() ?

Reply:
Refactored to avoid replicated code in the else case, which encompasses opset 9 and future opsets!